common/nacos_server/nacos_server.go

```diff
@@ -32,18 +34,29 @@ type NacosServer struct {
 	vipSrvRefInterMills int64
 }
 
-func NewNacosServer(serverList []constant.ServerConfig, httpAgent http_agent.IHttpAgent, timeoutMs uint64, endpoint string) (NacosServer, error) {
+func NewNacosServer(serverList []constant.ServerConfig, clientCfg constant.ClientConfig, httpAgent http_agent.IHttpAgent, timeoutMs uint64, endpoint string) (NacosServer, error) {
 	if len(serverList) == 0 && endpoint == "" {
 		return NacosServer{}, errors.New("both serverlist  and  endpoint are empty")
 	}
+
+	securityLogin := security.NewAuthClient(clientCfg, serverList, httpAgent)
```

Comment:
I think it's not need to create authClient when username or password is empty

Reply:
> I think it's not need to create authClient when username or password is empty

ok，I'll solve it

Reply:
> I think it's not need to create authClient when username or password is empty

I don't think so, but I think it's fair to say that inside the AuthClient it's just a matter of making a decision about whether or not the username is empty or whether or not you want to do the logic, because otherwise you're going to have to make a nil decision wherever you might be using AuthClient